cloudplatform/connectivity-apache-httpclient4/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultCsrfTokenRetriever.java

```diff
@@ -88,9 +89,13 @@ private Header retrieveCsrfTokenResponseHeader(
         // Additional headers
         headers.forEach(( k, values ) -> values.forEach(v -> csrfTokenRequest.addHeader(k, v)));
 
-        //The service path gets appended to the destination URI inside the execute request
+        // The service path gets appended to the destination URI inside the execute request
         final HttpResponse csrfResponse = httpClient.execute(csrfTokenRequest);
         final Header header = csrfResponse.getFirstHeader(X_CSRF_TOKEN_HEADER_KEY);
+
+        // Consume the HttpEntity and close the connection
+        EntityUtils.consume(csrfResponse.getEntity());
+
```

Comment:
```suggestion

        ((Closeable) httpClient).close();
```
Our `AbstractHttpClientFactory` always creates a `CloseableHttpClient` but we weakened the return to `HttpClient`.
Not sure if we reuse `HttpClient`s but couldn't find where

Reply:
We actually **must not** close the http client, otherwise we will break future connections which require this specific HTTP client